client/asset/btc/spv_wrapper.go

```diff
@@ -1157,8 +1158,54 @@ func (w *spvWallet) moveWalletData(backupDir string) error {
 	if err != nil {
 		return err
 	}
-	backupFolder := filepath.Join(backupDir, timeString)
-	return os.Rename(w.dir, backupFolder)
+
+	backupFolder := filepath.Join(backupDir, w.chainParams.Name, timeString)
+	// Copy wallet logs first. Even if there is an error, wallet files are
+	// still intact.
+	backupLogDir := filepath.Join(backupFolder, logDirName)
+	walletLogDir := filepath.Dir(w.logFilePath())
+	if err := copyDir(walletLogDir, backupLogDir); err != nil {
+		return err
+	}
+
+	walletBackupDir := filepath.Join(backupFolder, spvDir)
+	if err := os.Rename(w.dir, walletBackupDir); err != nil {
+		return err
+	}
+	return nil
+}
+
+// copyDir recursively copies the directories and files in source directory to
+// destination directory without preserving the original file permissions.
+func copyDir(srcDir, dstDir string) error {
+	dirFiles, err := os.ReadDir(srcDir)
+	if err != nil {
+		return err
+	}
+
+	if err := os.MkdirAll(dstDir, 0744); err != nil {
+		return err
+	}
+
+	for _, file := range dirFiles {
+		srcFile := filepath.Join(srcDir, file.Name())
+		dstFile := filepath.Join(dstDir, file.Name())
+		if file.IsDir() {
+			if err := copyDir(srcFile, dstFile); err != nil {
+				return err
+			}
+		} else if file.Type().IsRegular() {
+			logBytes, err := os.ReadFile(srcFile)
```

Comment:
We should never do `os.ReadFile` to copy files.  Please see the design here: https://github.com/decred/dcrdata/blob/master/stakedb/ticketpool.go#L174-L220

In addition to using a streaming buffered implementation of `copyFile`, it also `Stat`s the dest dir and only makes it if needed.  We can keep your current restriction that it must not exist, but note that please.

What your code has that the linked code does not seems to be recursion, which is fine to keep, but we should change how we copy the files.

Reply:
Alright, will look into the referenced code.